### PR TITLE
feat(dates): add formatYYYYMMDD func

### PR DESCRIPTION
**Pull request by benney-au-le**
Performance profiling svc-connection-synxis reveals that datelib.format function is run ~5000 times per request and accounts for a non-trivial amount of CPU time ~4%.

image

This new function leads to a measurable reduction in memory allocation requests / garbage collection.

using node --trace-gc flag
benchmark_2.js  = formatYYYYMMDD(d)
benchmark.js = format(d, 'yyyy-mm-dd')

```bash
~/s/svc-connection-synxis (main *% u=)
$ node --trace-gc benchmark_2.js 
[29638:0x52987f0]       93 ms: Scavenge 4.5 (4.8) -> 4.0 (5.8) MB, 1.3 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29638:0x52987f0]       99 ms: Scavenge 4.5 (5.8) -> 3.9 (6.3) MB, 1.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
~/s/svc-connection-synxis (main *% u=)
$ node --trace-gc benchmark.js 
[29683:0x4deb7f0]      101 ms: Scavenge 4.8 (5.1) -> 4.2 (6.1) MB, 1.1 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      107 ms: Scavenge 4.8 (6.1) -> 4.2 (6.6) MB, 1.5 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      120 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.5 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      130 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.3 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      139 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 3.6 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      150 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      156 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      164 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 1.0 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      169 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      175 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      183 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.3 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      190 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.3 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      195 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      203 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      207 ms: Scavenge 5.2 (6.6) -> 4.2 (6.6) MB, 0.3 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure 
[29683:0x4deb7f0]      215 ms: Scavenge 5.1 (6.6) -> 4.2 (6.6) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) task 
```

### CODE REVIEW

File: src/date/index.js [hunk 1]
```diff
@@ -57,6 +57,10 @@ function addDays(date, days) {
   return new Date(date.getTime() + days * 86400000)
 }
 
+function formatYYYYMMDD(date) {
```
Comment (mattswansonlux) [inline src/date/index.js:60]: Can you please add this function to index.d.ts?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@luxuryescapes/lib-global",
-  "version": "2.7.7",
+  "version": "2.8.7",
```
Comment (mattswansonlux) [inline package.json:3]: 2.8.0 would make more sense here

**Comment by mattswansonlux** [comment]
Great work here @benney-au-le. Love the flame graph!

File: package.json (hunk 2 above)
Comment (benney-au-le) [reply package.json]: oops rookie mistake.

File: src/date/index.js (hunk 1 above)
Comment (benney-au-le) [reply src/date/index.js:60]: yep done.

**Review by mattswansonlux — APPROVED** [review]
Looks good to me :+1: We might need @reb2020 to confirm if the complied files are still needed

**Pull request merged**